src/main/java/uk/co/bconline/ndelius/service/impl/UserServiceImpl.java

```diff
@@ -184,11 +184,9 @@ public Optional<User> getUserByStaffCode(String staffCode) {
 
 	@Override
 	public void addUser(User user) {
-		val dbFuture = runAsync(() -> userEntityService.save(transformer.mapToUserEntity(user, new UserEntity())), taskExecutor);
-		val ldapFuture = runAsync(() -> userEntryService.save(transformer.mapToUserEntry(user, new UserEntry())), taskExecutor);
-
 		try {
-			allOf(dbFuture, ldapFuture).join();
+			userEntityService.save(transformer.mapToUserEntity(user, new UserEntity()));
+			userEntryService.save(transformer.mapToUserEntry(user, new UserEntry()));
 		} catch (CancellationException | CompletionException e) {
```

Comment:
Could you update this catch clause too?
```suggestion
		} catch (Exception e) {
```